terragraph_planner/data/template.yaml

```diff
@@ -1,106 +1,104 @@
+  # be provided. Otherwise, all the intermediate topologies and the ILP problem
+  # files will be provided as well.
   DEBUG_MODE: False
 
-  # [String] The logger level that the planner will use, including NOTSET, DEBUG,
-  # INFO, WARNING, ERROR, CRITICAL. Please refer to the doc of Python logging for
-  # more details
+  # [String] The log level that the planner will use, including NOTSET, DEBUG,
+  # INFO, WARNING, ERROR, CRITICAL. Refer to the Python logging reference for
+  # more details.
   LOGGER_LEVEL: INFO
 
-  # [Optional][String] The file path of the log file. If it is null, the planner
-  # will not give the log in the file.
+  # [Optional][String] The file path of the log file. If it is null, log will
+  # not be recorded to file.
   LOG_FILE: null
 
-  # [Bool] Whether or not the planner prints log to the stderr.
+  # [Bool] Enable logging to stderr.
   LOG_TO_STDERR: True
 
 DATA:
-  # [String] The area that planning will be restricted to. The file
-  # can contain only one polygon. File type: KML or KMZ
+  # [Required][String] The area of interest of the network plan. File type:
+  # KML/KMZ.
   BOUNDARY_POLYGON_FILE_PATH: boundary.kml
 
-  # [List of String] A list of GeoTIFF files containing
-  # the height of canopy (terrain elevation + height of any clutter above
-  # ground) (.tif).
-
+  # [List of String] A list of GeoTIFF files containing the height of canopy
+  # (terrain elevation + height of any clutter above ground) File type: TIF.
   DSM_FILE_PATHS:
     - dsm.tif
 
-  # [String] DTM_FILE_PATH contains the digital terrain
-  # model, which represents the elevation of the earth's surface. File type: TIF
-  # DHM_FILE_PATH contains the digital height model, which represents the height
-  # of aboveground objects. File type: TIF
-
+  # [String] DTM_FILE_PATH contains the digital terrain model, which represents
+  # the elevation of the earth's surface. File type: TIF.
+  # DHM_FILE_PATH contains the digital height model, which represents the
+  # height of aboveground objects. File type: TIF.
   DTM_FILE_PATH: dtm.tif
   DHM_FILE_PATH: dhm.tif
 
-  # [String] The files that contain the geometric locations and
-  # attributes of geographic features, such as polygons and heights of
-  # buildings. Get data from a zipped shape file or a KML/KMZ file.
-  # If a zipped shape file is given, They must be uploaded as a ZIP file containing
-  # SHP, SHX, PRJ and DBF shapefiles that are not in a nested directory. For
-  # example, the directory for the SHP file should follow "shapefile.zip/
-  # shp_file.shp" and not "shapefile.zip/some_folder/shp_file.shp."
-
+  # [String] The files that contain the geometric locations and attributes of
+  # geographic features, such as polygons and heights of buildings. File type:
+  # zipped shape file or a KML/KMZ file. If a zipped shape file is given, it
+  # must be uploaded as a ZIP file containing SHP, SHX, PRJ and DBF shapefiles
```

Comment:
I'm assuming this means it can contain any combination of SHP, SHX, PRJ, and DBF shapefiles? What if it is an empty zip? I don't know if it is worth clarifying, but just thought I'd mention it.

Reply:
I think it must be all of them, right @wuweicaifb?

Reply:
Although the planner only explicitly checks the following two rules for shp file:
1. SHP must exist and only one exist
2. no subfolders

Other files, including SHX, PRJ and DBF are still needed for the zip file. Otherwise the plan will fail. For example, if PRJ is missing, there would be no projection information, which is required and checked by the planner too. SHX is the index file and dbf is the dBase file, which are also needed to load a valid shp file (otherwise the plan will fail). When a SHP file is created by gdal lib or GIS tools, others are created automatically.

More details: 
https://www.esri.com/content/dam/esrisites/sitecore-archive/Files/Pdfs/library/whitepapers/pdfs/shapefile.pdf

Reply:
Daniel, I'll leave it up to you what level of detail you want to put here. I'll approve this, but you can update this last thing if you want to then merge.